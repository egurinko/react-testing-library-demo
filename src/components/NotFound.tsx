import React from 'react';

const NotFound: React.FC<NoProps> = () => (
  <div>
    <h1>Not Found</h1>
    <h1>404</h1>
  </div>
);

const memoed = React.memo(NotFound);
export { memoed as NotFound };
