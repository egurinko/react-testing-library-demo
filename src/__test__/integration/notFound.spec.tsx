import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';
import NotFound from '../../components/NotFound';

let documentBody: RenderResult;

describe('<NotFound />', () => {
  beforeEach(() => {
    documentBody = render(<NotFound />);
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
