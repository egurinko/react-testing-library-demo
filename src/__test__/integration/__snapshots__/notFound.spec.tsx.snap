// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> matches snapshot 1`] = `
<body>
  <div>
    <div>
      <h1>
        Not Found
      </h1>
      <h1>
        404
      </h1>
    </div>
  </div>
</body>
`;
